lib/Sema/TypeCheckProtocolInference.cpp

```diff
@@ -188,22 +188,27 @@ AssociatedTypeInference::inferTypeWitnessesViaValueWitnesses(
     // Invalid case.
     if (extendedNominal == nullptr)
       return true;
-
+    
+    // Validate the nominal type being extended.
+    tc.validateDecl(extendedNominal);
+    if (extendedNominal->isInvalid())
+      return true;
+    
     // Assume unconstrained concrete extensions we found witnesses in are
     // always viable.
     if (!isa<ProtocolDecl>(extendedNominal))
       return !extension->isConstrainedExtension();
```

Comment:
isConstrainedExtension() calls getGenericSignature(). It could probably just be a syntactic property instead, by checking for the presence of a parsed where clause.